linkerd/proxy/spire-client/src/api.rs

```diff
@@ -0,0 +1,208 @@
+use futures::prelude::*;
+use linkerd_error::{Error, Recover, Result};
+use linkerd_exp_backoff::{ExponentialBackoff, ExponentialBackoffStream};
+use linkerd_identity::DerX509;
+use linkerd_identity::Id;
+use linkerd_proxy_http as http;
+use linkerd_tonic_watch::StreamWatch;
+use spiffe_proto::client::{
+    self as api, spiffe_workload_api_client::SpiffeWorkloadApiClient as Client,
+};
+use std::collections::HashMap;
+use tower::Service;
+use tracing::error;
+
+#[derive(Clone)]
+pub struct Svid {
+    pub(super) spiffe_id: Id,
+    pub(super) leaf: DerX509,
+    pub(super) private_key: Vec<u8>,
```

Comment:
This is a bit of yakshave that we can do as a followup, but I think we really ought to have a linkerd_identity::PrivateKey type so we're not passing around sensitive byte arrays everywhere.

Why does the private key need to be crate-visible?

Reply:
Ah, I see this is for process_svid -- what if we moved process_svid into this module and made _that_ `pub(super)` -- this gives us a narrower module boundary where we can make all of the Svid stuff private